src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -27,1120 +43,848 @@ macro_rules! sl {
-            write_file(dir, OOM_CONTROL, 1)?;
+        match t[1].trim().parse::<u64>() {
+            Ok(v) => {
+                m.insert(t[0].to_string(), v);
+            }
+            Err(_) => {}
         }
-
-        Ok(())
     }
-}
-
-fn get_exist_memory_data(dir: &str, sub: &str) -> Result<Option<MemoryData>> {
-    let res = match get_memory_data(dir, sub) {
-        Err(ref e) if io_error_kind_eq(e, std::io::ErrorKind::NotFound) => None,
 
-        Ok(r) => Some(r),
-
-        Err(e) => return Err(anyhow!(e)),
-    };
-
-    Ok(res)
+    m
 }
 
-fn get_memory_data(dir: &str, sub: &str) -> Result<MemoryData> {
-    let base = "memory";
-    let (fusage, fmax_usage, ffailcnt, flimit) = if sub.is_empty() {
-        (
-            format!("{}.{}", base, MEM_USAGE),
-            format!("{}.{}", base, MEM_MAX_USAGE),
-            format!("{}.{}", base, MEM_FAILCNT),
-            format!("{}.{}", base, MEM_LIMIT),
-        )
-    } else {
-        (
-            format!("{}.{}.{}", base, sub, MEM_USAGE),
-            format!("{}.{}.{}", base, sub, MEM_MAX_USAGE),
-            format!("{}.{}.{}", base, sub, MEM_FAILCNT),
-            format!("{}.{}.{}", base, sub, MEM_LIMIT),
-        )
-    };
-
-    let usage = get_param_u64(dir, fusage.as_str())?;
-    let max_usage = get_param_u64(dir, fmax_usage.as_str())?;
-    let failcnt = get_param_u64(dir, ffailcnt.as_str())?;
-    let limit = get_param_u64(dir, flimit.as_str())?;
+pub const NANO_PER_SECOND: u64 = 1000000000;
+pub const WILDCARD: i64 = -1;
 
-    Ok(MemoryData {
-        usage,
-        max_usage,
-        failcnt,
-        limit,
-        unknown_fields: UnknownFields::default(),
-        cached_size: CachedSize::default(),
-    })
-}
+lazy_static! {
+    pub static ref CLOCK_TICKS: f64 = {
+        let n = unsafe { libc::sysconf(libc::_SC_CLK_TCK) };
 
-impl Memory {
-    fn get_stats(&self, dir: &str) -> Result<MemoryStats> {
-        let h = get_param_key_u64(dir, MEMORY_STAT)?;
-        let cache = *h.get("cache").unwrap();
-        info!(sl!(), "cache");
-
-        let value = get_param_u64(dir, MEM_HIERARCHY)?;
-        let use_hierarchy = if value == 1 { true } else { false };
-
-        info!(sl!(), "hierarchy");
-
-        // gte memory datas
-        let usage = SingularPtrField::from_option(get_exist_memory_data(dir, "")?);
-        let swap_usage = SingularPtrField::from_option(get_exist_memory_data(dir, "memsw")?);
-        let kernel_usage = SingularPtrField::from_option(get_exist_memory_data(dir, "kmem")?);
-
-        Ok(MemoryStats {
-            cache,
-            usage,
-            swap_usage,
-            kernel_usage,
-            use_hierarchy,
-            stats: h,
-            unknown_fields: UnknownFields::default(),
-            cached_size: CachedSize::default(),
-        })
-    }
-}
+        n as f64
+    };
+    pub static ref DEFAULT_ALLOWED_DEVICES: Vec<LinuxDeviceCgroup> = {
+        let mut v = Vec::new();
+        v.push(LinuxDeviceCgroup {
```

Comment:
Maybe add a short comment for each of these to indicate the kind of devices we are dealing with.

Reply:
Yes, it will be more readable.